catalog/dags/common/popularity/sql.py

```diff
@@ -439,6 +439,54 @@ def create_media_view(
         postgres.run(create_audioset_view_query())
 
 
+def get_providers_with_popularity_data_for_media_type(
+    postgres_conn_id: str,
+    media_type: str = IMAGE,
+    constants_view: str = IMAGE_POPULARITY_CONSTANTS_VIEW,
```

Comment:
Should we maybe just make these required variables, so we don't have to add additional logic for future media types?

Or if the concern is that this is a nested function and we don't want to hand in the pop constants view through the entire stack, maybe we could use a global mapping from media type to constants view and look up from there? Or use the media type to format the name of the constants view?

Reply:
I opened #2678 to refactor all of `popularity/sql.py` to do this 💯